lib/stdlib/src/supervisor.erl

```diff
@@ -1345,6 +1444,16 @@ validRestartType(temporary)   -> true;
 validRestartType(transient)   -> true;
 validRestartType(RestartType) -> throw({invalid_restart_type, RestartType}).
 
+validSignificant(true, permanent, _AutoShutdown) ->
+    throw({invalid_significant, true});
```

Comment:
Should be no problem I think.

Reply:
@garazdawi going by your suggestion, both

```
restart => permanent,
significant => true
```
and
```
significant => permanent_significant_child
```
would result in the same error `{invalid_significant, permanent_significant_child}`.

The second case is not very likely I admit, but still I would rather throw just `permanent_significant_child` in the first case (an invalid combination of `restart => permanent` and `significant => true`) and `{invalid_significant, permanent_significant_child}` in the second case (invalid value `permanent_significant_child` for `significant`).

For the other invalid combination, `auto_shutdown => never` and `significant => true`, I would use error reason `significant_child_auto_shutdown_never` as error reason. A bit clumsy, but it's the best I can come up with. What do you think?

Reply:
Brainstorming: Maybe instead of `invalid_significant` we should have ìnvalid_option_combination` and then return a map, or even a string describing the error? I haven't checked if the supervisor API allows this to be returned by the error. These errors are really only meant to be printed and consumed by a human.

It is quite a detour from the way other errors are presented in supervisor, so maybe the extra clarity is not worth the in-consistency...

Reply:
I think the supervisor API allows anything as error, at least the specs for `start_link`, `start_child` and `check_childspecs` all have `term()` in the error tuples. I'm fine with maps and even strings, but as you say, it _is_ quite different from everything else in `supervisor`...

Reply:
So what shall we do here? I'm almost in favor of leaving it as it is now, at least for RC3 :/

Reply:
does @HansN or @IngelaAndin have any opinions?

Reply:
> For me we could leave this till after RC3.

let us leave it as is and make sure we come up with something better for the final release.